### PR TITLE
Unify API on seed keyword for random seeds / generator

### LINKED ISSUE #5359: API consistency: seed vs random_seed vs sample_seed
## Description

See [this comment](https://github.com/scikit-image/scikit-image/pull/5357#issuecomment-827075682) from @grlee77:

> it seems we were not very consistent in naming the argument seed vs. random_seed across the library (I think I also saw sample_seed in one place). Making this more consistent is something to consider separately for the 1.0 API.

For 1.0, we should make sure we pick one specific name. My vote is for `random_seed=`.

### PR DESCRIPTION
**Pull request by grlee77**
## Description

closes #5359

Unify on use of `seed` in the public API. Currently had a mixture of (`sample_seed`, `seed`, `random_seed`, `random_state`)

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/morphology/_skeletonize.py [hunk 1]
```diff
@@ -372,13 +374,12 @@ def medial_axis(image, mask=None, return_distance=False, *, random_state=None):
         value in `mask` are used for computing the medial axis.
     return_distance : bool, optional
         If true, the distance transform is returned as well as the skeleton.
-    random_state : {None, int, `numpy.random.Generator`}, optional
-        If `random_state` is None the `numpy.random.Generator` singleton is
+    seed : {None, int, `numpy.random.Generator`}, optional
+        If `seed` is None the `numpy.random.Generator` singleton is used.
+        If `seed` is an int, a new ``Generator`` instance is used, seeded with
+        `seed`.
+        If `seed` is already a ``Generator`` instance then that instance is
```
Comment (mkcor) [inline skimage/morphology/_skeletonize.py:381]:
```suggestion
        If `seed` is already a ``Generator`` instance, then that instance is
```

**Review by mkcor — APPROVED** [review]
Arguably the depreciation cycle is getting squeezed here (`deprecated_version='0.20', removed_version='1.0'`). I'm +1 in spite of this. /cc @scikit-image/core

**Review by stefanv — APPROVED** [review]

**Comment by soupault** [comment]
The PR looks good 👍🏻 . Please, rebase onto the latest commit to rerun the tests.
It would be great also to have a reminder in `TODO.txt` to remove the deprecation wrappers in v1.0.

**Comment by stefanv** [comment]
@grlee77 This one is good to go, right? Do you want to rebase and get it into the forthcoming release?

**Comment by grlee77** [comment]
I was slow for 0.20, but the conflicts have now been fixed. Will need one more commit to update the versions listed for the warning from `deprecate_kwarg`.

**Review by jarrodmillman — APPROVED** [review]

File: skimage/feature/brief.py [hunk 2]
```diff
@@ -116,8 +116,10 @@ class BRIEF(DescriptorExtractor):
 
     """
 
+    @deprecate_kwarg({'sample_seed': 'seed'}, deprecated_version='0.20',
```
Comment (jarrodmillman) [inline skimage/feature/brief.py:119]: Should this be `deprecated_version='0.21``?

**Comment by jarrodmillman** [comment]
Shouldn't we add a note to `TODO.txt`?

**Pull request merged**